### PR TITLE
[JENKINS-43802] Implementing retrieve/retrieveRevisions with an Item context

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/scm-api-plugin/pull/65.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -142,7 +142,7 @@ THE SOFTWARE.
     <workflow-scm-step.version>2.6</workflow-scm-step.version>
     <workflow-aggregator.version>2.5</workflow-aggregator.version>
     <findbugs.failOnError>false</findbugs.failOnError> <!-- TODO still have 27 left -->
-    <scm-api-plugin.version>2.2.6</scm-api-plugin.version>
+    <scm-api-plugin.version>2.5.0-rc464.915933672136</scm-api-plugin.version> <!-- TODO https://github.com/jenkinsci/scm-api-plugin/pull/65 -->
```
Comment (bitwiseman) [inline pom.xml]:
```suggestion
        <scm-api-plugin.version>2.6.2</scm-api-plugin.version>
```

File: pom.xml [hunk 2]
```diff
@@ -142,7 +142,7 @@ THE SOFTWARE.
     <workflow-scm-step.version>2.6</workflow-scm-step.version>
     <workflow-aggregator.version>2.5</workflow-aggregator.version>
     <findbugs.failOnError>false</findbugs.failOnError> <!-- TODO still have 27 left -->
-    <scm-api-plugin.version>2.2.6</scm-api-plugin.version>
+        <scm-api-plugin.version>2.6.2</scm-api-plugin.version>
```
Comment (bitwiseman) [inline pom.xml]:
```suggestion
    <scm-api-plugin.version>2.6.3</scm-api-plugin.version>
```

**Comment by jglick** [comment]
CI issue again…

**Comment by jglick** [comment]
Now we are [stuck in GitHub](https://ci.jenkins.io/job/Plugins/job/subversion-plugin/indexing/events). Sigh, maybe time to switch to Prow.

**Comment by jglick** [comment]
Now we are stuck on [INFRA-1633](https://issues.jenkins-ci.org/browse/INFRA-1633).

**Pull request merged**